src/compiler/compile/render_dom/wrappers/Element/index.ts

```diff
@@ -289,7 +289,10 @@ export default class ElementWrapper extends Wrapper {
 
 		// insert static children with textContent or innerHTML
 		const can_use_textcontent = this.can_use_textcontent();
```

Comment:
Shouldn't this function return `false` if the node is a `template` element? Instead of modifying the code below.

We also use snake_case for internal variables.

Reply:
the explicit early check is needed to handle the ambiguity of 
the or condition `(can_use_innerhtml || can_use_textcontent)` .

which will fail this case:

```js
<template>123</template> // can_use_innerhtml:true || can_use_textcontent:false
```

```js

!is_template_with_text_content         // short here
 &&  !this.node.namespace 
 && (this.can_use_innerhtml || can_use_textcontent)   // <-- we should not get here for template
 && this.fragment.nodes.length > 0
```